google/cloud/sql/connector/asyncpg.py

```diff
@@ -48,6 +59,10 @@ async def connect(
         raise ImportError(
             'Unable to import module "asyncpg." Please install and try again.'
         )
+
+    async def async_sock_func() -> ssl.SSLSocket:
+        return await asyncio.to_thread(sock_func, ip_address, ctx)
```

Comment:
`asyncio.to_thread` was added in Python 3.9. Will have to implement this on my own to support Python 3.8.